### PR TITLE
Webex InMeeting Component

### PR DESCRIPTION
**Pull request by akoushke**

### CODE REVIEW

File: src/components/WebexInMeeting/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Webex Interstitial Meeting Component
+
+Webex InMeeting component displays the [Webex Local Media](../WebexLocalMedia) on top of [Webex Remote Media](../WebexRemoteMedia) component all together for a meeting.
```
Comment (adamweeks) [inline src/components/WebexInMeeting/README.md]:
```suggestion
Webex InMeeting component displays the [Webex Local Media](../WebexLocalMedia) component on top of the [Webex Remote Media](../WebexRemoteMedia) component for the in meeting media experience.
```

File: src/components/WebexInMeeting/WebexInMeeting.stories.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import {WebexInMeeting, WebexDataProvider} from '..';
+
+// Setup for the stories
+const stories = storiesOf('Webex InMeeting', module);
+const webexAdapter = new WebexJSONAdapter(jsonData);
+const wrapperStyle = {height: '500px', width: '800px', border: '1px solid black'};
+
+stories.add('remote and local media enabled', () => {
+  jsonData.meetings.remoteMedia.remoteAudio = {};
```
Comment (adamweeks) [inline src/components/WebexInMeeting/WebexInMeeting.stories.js]: Aren't these usually stored in the data itself?

File: src/components/WebexInMeeting/WebexInMeeting.test.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import WebexInMeeting from '../WebexInMeeting/WebexInMeeting';
+
+jest.mock('../hooks/useMeeting');
+
+describe('Webex InMeeting component', () => {
+  describe('snapshot', () => {
+    test('matches snapshot of meeting destination', () => {
```
Comment (adamweeks) [inline src/components/WebexInMeeting/WebexInMeeting.test.js]:
```suggestion
    test('matches snapshot of meeting', () => {
```

File: src/components/WebexInMeeting/WebexInMeeting.stories.js (hunk 2 above)
Comment (akoushke) [reply src/components/WebexInMeeting/WebexInMeeting.stories.js]: they are but coming from the other state to this one would not reset the properties to default. basically, it does not re-read the data.

File: src/components/WebexInMeeting/WebexInMeeting.stories.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    <div style={wrapperStyle}>
+      <WebexDataProvider adapter={webexAdapter}>
+        <WebexInMeeting meetingID="remoteMedia" />
+      </WebexDataProvider>
+    </div>
+  );
+});
+
+stories.add('only remote media enabled', () => {
+  jsonData.meetings.remoteMedia.remoteAudio = {};
```
Comment (adamweeks) [inline src/components/WebexInMeeting/WebexInMeeting.stories.js]: You could use `jsonData.meetings['remoteAudio&Video']` here and then you wouldn't need to reset the other one.

Comment (akoushke) [reply src/components/WebexInMeeting/WebexInMeeting.stories.js]: interstitial doing the same. I guess we need to make the changed in that file as well.

File: src/components/WebexInMeeting/README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+Webex InMeeting component displays the [Webex Local Media](../WebexLocalMedia) component on top of the [Webex Remote Media](../WebexRemoteMedia) component for the in meeting media experience.
+
+<p align="center">
+  <img src="./WebexInMeeting.png" alt="Default Webex InMeeting" />
+</p>
+
+## Preview
+
+To see all the different possible states of the Webex Interstitial Meeting component, you can run our Storybook:
```
Comment (lalli-flores) [inline src/components/WebexInMeeting/README.md]:
```suggestion
To see all the different possible states of the Webex In-Meeting component, you can run our Storybook:
```

File: src/components/WebexInMeeting/README.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
+# Webex InMeeting Component
+
+Webex InMeeting component displays the [Webex Local Media](../WebexLocalMedia) component on top of the [Webex Remote Media](../WebexRemoteMedia) component for the in meeting media experience.
```
Comment (lalli-flores) [inline src/components/WebexInMeeting/README.md]:
```suggestion
Webex In-Meeting component displays the [Webex Local Media](../WebexLocalMedia) component on top of the [Webex Remote Media](../WebexRemoteMedia) component for the in meeting media experience.
```

File: src/components/WebexInMeeting/README.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
+# Webex InMeeting Component
```
Comment (lalli-flores) [inline src/components/WebexInMeeting/README.md]:
```suggestion
# Webex In-Meeting Component
```

File: src/components/WebexInMeeting/README.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+## Embed
+
+1.  Create a component adapter from which the data will be retrieved (See [adapters](../../adapters)). For instance:
+
+    ```js
+    const jsonAdapter = new WebexJSONAdapter(jsonData);
+    ```
+
+2.  Create a component instance by passing the person ID as a string and
```
Comment (lalli-flores) [inline src/components/WebexInMeeting/README.md]:
```suggestion
2.  Create a component instance by passing the meeting ID as a string and
```

File: src/components/WebexInMeeting/WebexInMeeting.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {WebexLocalMedia, WebexRemoteMedia} from '..';
+
+import './WebexInMeeting.css';
+
+/**
+ * Webex InMeeting component displays the remote stream plus
```
Comment (lalli-flores) [inline src/components/WebexInMeeting/WebexInMeeting.js]:
```suggestion
 * Webex In-Meeting component displays the remote stream plus
```

File: src/components/WebexInMeeting/WebexInMeeting.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+ * the local stream at the bottom right corner.
+ *
+ * @param {object} props
+ * @returns {object} JSX of the component
+ */
+export default function WebexInMeeting({meetingID}) {
+  return (
+    <div className="in-meeting">
+      <React.Fragment>
```
Comment (lalli-flores) [inline src/components/WebexInMeeting/WebexInMeeting.js]: If you're enclosing the media components in a div, the `Fragment` is unnecessary, please remove it

File: src/components/WebexInMeeting/WebexInMeeting.css [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+.in-meeting .local-media {
+  position: absolute;
+  bottom: 25%;
+  right: 3%;
+  width: 20%;
+  height: 20%;
+  z-index: 1;
+  background: white;
```
Comment (lalli-flores) [inline src/components/WebexInMeeting/WebexInMeeting.css]: Why are making the background white?

File: src/components/WebexInMeeting/WebexInMeeting.css [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+.in-meeting .remote-media {
+  position: relative;
+  width: 100%;
+  height: 100%;
+}
+
+.in-meeting .local-media {
+  position: absolute;
+  bottom: 25%;
+  right: 3%;
```
Comment (lalli-flores) [inline src/components/WebexInMeeting/WebexInMeeting.css]: How did you arrive at this percentages?

File: src/components/index.js [hunk 13]
```diff
@@ -3,6 +3,7 @@ export {default as WebexActivity} from './WebexActivity/WebexActivity';
 export {default as WebexActivityStream} from './WebexActivityStream/WebexActivityStream';
 export {default as WebexDataProvider, AdapterContext} from './WebexDataProvider/WebexDataProvider';
 export {default as WebexInterstitialMeeting} from './WebexInterstitialMeeting/WebexInterstitialMeeting';
+export {default as WebexInMeeting} from './WebexInMeeting/WebexInMeeting';
```
Comment (lalli-flores) [inline src/components/index.js]: If we're importing alphabetically, this should come before `WebexInterstitialMeeting`

**Comment by lalli-flores** [comment]
> Approved, but we have a bug with the local media avatar sizing. It will not shrink with the local media size and overlaps now:
> 
> image

@adamweeks , if you found a bug, we shouldn't approve the PR. Why would we introduce bugs into the code base?

**Comment by akoushke** [comment]
> > Approved, but we have a bug with the local media avatar sizing. It will not shrink with the local media size and overlaps now:
> > image
> 
> @adamweeks , if you found a bug, we shouldn't approve the PR. Why would we introduce bugs into the code base?

this styling bug is from a different component and took some notes to create a JIRA and get back to it after we finish the meeting component. I have a list of action items that have less priorities ATM.

**Comment by lalli-flores** [comment]
> > > Approved, but we have a bug with the local media avatar sizing. It will not shrink with the local media size and overlaps now:
> > > image
> > 
> > 
> > @adamweeks , if you found a bug, we shouldn't approve the PR. Why would we introduce bugs into the code base?
> 
> this styling bug is from a different component and took some notes to create a JIRA and get back to it after we finish the meeting component. I have a list of action items that have less priorities ATM.

Ah, just re-read @adamweeks comment. My bad, I misunderstood. As long as we tracked the known issue, it's cool

**Comment by akoushke** [comment]
> > > > Approved, but we have a bug with the local media avatar sizing. It will not shrink with the local media size and overlaps now:
> > > > image
> > > 
> > > 
> > > @adamweeks , if you found a bug, we shouldn't approve the PR. Why would we introduce bugs into the code base?
> > 
> > 
> > this styling bug is from a different component and took some notes to create a JIRA and get back to it after we finish the meeting component. I have a list of action items that have less priorities ATM.
> 
> Ah, just re-read @adamweeks comment. My bad, I misunderstood. As long as we tracked the known issue, it's cool

nothing to reply here. just wanted another copy of the screenshot in this PR :)

**Comment by akoushke** [comment]
> What happens in this component when there is no local media or it is muted?

the localMedia will display the the avatar instead.

File: src/components/WebexInMeeting/WebexInMeeting.css (hunk 11 above)
Comment (akoushke) [reply src/components/WebexInMeeting/WebexInMeeting.css]: the background is transparent and thats not what we want.

Comment (akoushke) [reply src/components/WebexInMeeting/WebexInMeeting.css]: so I set it to white.

File: src/components/WebexInMeeting/WebexInMeeting.css (hunk 12 above)
Comment (akoushke) [reply src/components/WebexInMeeting/WebexInMeeting.css]: no idea. just looks good. any better approach? im trying to leave some room for the controls and other streams at the bottom.

File: src/components/index.js (hunk 13 above)
Comment (akoushke) [reply src/components/index.js]: :|

File: src/components/WebexInMeeting/WebexInMeeting.css (hunk 11 above)
Comment (akoushke) [reply src/components/WebexInMeeting/WebexInMeeting.css]: the localMedia component needs some styling works. I removed it form this PR. there will be a ticket for styling the localMedia

File: src/components/WebexInMeeting/WebexInMeeting.css (hunk 12 above)
Comment (lalli-flores) [reply src/components/WebexInMeeting/WebexInMeeting.css]: We might have to do some more testing in different screen sizes

File: src/components/WebexInMeeting/WebexInMeeting.css [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+  width: 20%;
+  height: 20%;
+  z-index: 1;
+  border: solid 1px black;
```
Comment (lalli-flores) [inline src/components/WebexInMeeting/WebexInMeeting.css]: I know I didn't make a comment about this before, but not sure we should add this border

**Pull request merged**